src/main/kotlin/com/cognifide/gradle/aem/instance/content/Node.kt

```diff
@@ -0,0 +1,110 @@
+package com.cognifide.gradle.aem.instance.content
+
+import com.cognifide.gradle.aem.common.http.RequestException
+import com.cognifide.gradle.aem.instance.InstanceSync
+import com.jayway.jsonpath.DocumentContext
+import com.jayway.jsonpath.JsonPath
+import net.minidev.json.JSONArray
+import org.apache.jackrabbit.vault.util.JcrConstants
+
+class Node private constructor() {
+
+    lateinit var sync: InstanceSync
+    lateinit var path: String
+    lateinit var name: String
+    lateinit var document: DocumentContext
+
+    fun children(): Iterator<Node> {
+        return sync.get("$path.harray.1.json") { asJson(it) }
+                .read<JSONArray>("__children__")
+                .map { props -> loadChild(sync, path, props as MutableMap<*, *>) }
+                .iterator()
+    }
+
+    fun get(propName: String): Any = document.read(propName) as Any
+    fun getString(propName: String): String = document.read(propName) as String
+    fun json(): String = document.jsonString()
+
+    fun getJcrTitle(): String = getString(JcrConstants.JCR_TITLE)
+    fun getJcrCreated() = get(JcrConstants.JCR_CREATED)
+    fun getJcrCreatedBy(): String = getString(JcrConstants.JCR_CREATED_BY)
+    fun getJcrDescription(): String = getString(JcrConstants.JCR_DESCRIPTION)
+    fun getJcrPrimaryType(): String = getString(JcrConstants.JCR_PRIMARYTYPE)
+    fun getSlingResourceType(): String = getString("sling:resourceType")
+    fun getCqAllowedTemplates() = get("cq:allowedTemplates")
```

Comment:
I am not sure about listing "cq" properties here.... how to do it right / provide full list... maintaining that could be pain in the ass later ; GAP should never rely on concrete AEM version... and such properties are AEM specific / newer AEM's are introducing new properties with "cq:" prefix

Reply:
`cq`, `sling` removed